azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_provider.py

```diff
@@ -88,11 +89,21 @@ def on_message_callback(client, userdata, mqtt_message):
                 logger.error("Unexpected error calling on_mqtt_message_received")
                 logger.error(traceback.format_exc())
 
+        def on_unsubscribe_callback(client, userdata, mid):
+            logger.info("UNSUBACK received for %s", str(mid))
+            # TODO: how to do failure?
```

Comment:
># TODO: how to do failure? [](start = 12, length = 26)

I don't think this can "fail". either you get an UNSUBACK and it worked, or you don't, right?

Reply:
I think it is referring to except ? all these callbacks probably mean how to handle the exception block...I did not want to break the pattern hence kept the same TODO

---
In reply to: [258251395](https://github.com/Azure/azure-iot-sdk-python-preview/pull/50#discussion_r258251395) [](ancestors = 258251395)